### PR TITLE
Issue 46122: Pipeline actions don't return to the starting URL, instead redirect to folder homepage

### PR DESCRIPTION
**Pull request by labkey-jeckels**
#### Rationale
We used to get users back to the folder tab they started via .lastFilter. That not longer remembers the tab name, and users have requested that this particular workflow continue to work the same.

#### Changes
* Pull the returnUrl through the full sequence and send the user back there

### CODE REVIEW

File: filecontent/webapp/File/panel/Browser.js [hunk 1]
```diff
@@ -275,8 +275,13 @@ Ext4.define('File.panel.Browser', {
          * @private
          */
         _getActions : function(cb, containerPath, scope) {
+            var params = {path : scope.getFullFolderOffset() };
+            var returnUrl = LABKEY.ActionURL.getParameter('returnUrl');
```
Comment (labkey-nicka) [inline filecontent/webapp/File/panel/Browser.js]: There is another function for this as well. `LABKEY.ActionURL.getReturnUrl()`. Same feedback for usage in `analyze.jsp`.

File: api/src/org/labkey/api/pipeline/file/FileAnalysisTaskPipeline.java [hunk 2]
```diff
@@ -74,7 +75,7 @@ interface FilePathFilter extends FileFilter, DirectoryStream.Filter<Path>
     FilePathFilter getInitialFileTypeFilter();
 
     @NotNull
-    URLHelper getAnalyzeURL(Container c, String path);
+    URLHelper getAnalyzeURL(Container c, String path, ReturnURLString parsedReturnUrl);
```
Comment (labkey-nicka) [inline api/src/org/labkey/api/pipeline/file/FileAnalysisTaskPipeline.java]: Denote `@Nullable` on `parsedReturnUrl`.

**Review by labkey-nicka — APPROVED** [review]
Looks good. I presume it's known why `.lastFilter` no longer supports this and/or is not still supported.

**Comment by labkey-jeckels** [comment]
@labkey-nicka here's the change that stopped returning to the original folder tab https://github.com/LabKey/platform/pull/3426

**Pull request merged**